platforms/software/dependency-management/src/test/groovy/org/gradle/internal/component/external/model/DefaultModuleComponentSelectorTest.groovy

```diff
@@ -163,21 +150,13 @@ class DefaultModuleComponentSelectorTest extends Specification {
         selector.versionConstraint.strictVersion == ''
         selector.versionConstraint.rejectedVersions == []
         selector.attributes.isEmpty()
-        selector.requestedCapabilities == [new DefaultImmutableCapability("org", "blah", "1.5")]
+        selector.capabilitySelectors == capabilities
+        selector.requestedCapabilities[0] == ((DefaultSpecificCapabilitySelector) capabilities[0]).backingCapability
+        selector.requestedCapabilities[1] == new DefaultImmutableCapability("some-group", "some-name-foo", "1.0")
```

Comment:
💅 This seems odd.  Why verify the backingCapability?  Isn't that an implementation detail?  It's just `capabilities[0]` here that matters, right?

Reply:
capabilities[0] is a capability selector. `selector.requestedCapabilities[0] == capabilities[0]` would fail since a CapabilitySelector != Capability. 

I've renamed `capabilities` to `capabilitySelectors` to reflect that